galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.selenium.manager/src/main/java/dev/galasa/selenium/internal/SeleniumEnvironment.java

```diff
@@ -96,35 +102,37 @@ private String allocateSlot() throws ResourceUnavailableException, SeleniumManag
 		String slotName = "";
 		int currentSlots = 0;
 		try {
-			slots = dss.get(slotKey);
-			if (slots != null) {
-				currentSlots = Integer.valueOf(slots);
-			}
-			
-			if (currentSlots >= SeleniumDriverMaxSlots.get()) {
-				throw new ResourceUnavailableException("Failed to provsion. No slots avilable");
-			}			
-			
-			String slotNamePrefix = "SeleniumSlot_" + this.runName + "_";
-			
-			int counter = 0;
 			while ("".equals(slotName)) {
```

Comment:
change this to a for loop 1->1000  if you don't have a slotname by 1000 attempts,  throw an exception and report the last exception that was received from the performactions that prevented the slot